src/Microsoft.Framework.FileSystemGlobbing/Internal/PathSegments/CurrentPathSegment.cs

```diff
@@ -3,11 +3,11 @@
 
 using System;
 
-namespace Microsoft.Framework.FileSystemGlobbing.Infrastructure
+namespace Microsoft.Framework.FileSystemGlobbing.Internal.PathSegments
 {
-    public abstract class PatternSegment
+    public class CurrentPathSegment : IPathSegment
     {
-        public virtual bool TestMatchingSegment(string value, StringComparison comparisonType)
+        public bool Match(string value, StringComparison comparisonType)
         {
             return false;
```

Comment:
Please check the latest commit which dealing with the `CurrentPathSegment`. In short the `Match` method is never expected to be invoked on `CurrentPathSegment` and `RecursivePathSegment` since they're not added to the path list.